app/src/main/java/io/gnosis/safe/ui/safe/selection/SafeSelectionDialog.kt

```diff
@@ -0,0 +1,97 @@
+package io.gnosis.safe.ui.safe.selection
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import io.gnosis.safe.HeimdallApplication
+import io.gnosis.safe.R
+import io.gnosis.safe.databinding.DialogSafeSelectionBinding
+import io.gnosis.safe.di.components.DaggerViewComponent
+import io.gnosis.safe.di.modules.ViewModule
+import io.gnosis.safe.ui.base.BaseBottomSheetDialogFragment
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.safe.SafeOverviewNavigationHandler
+import javax.inject.Inject
+
+class SafeSelectionDialog : BaseBottomSheetDialogFragment<DialogSafeSelectionBinding>() {
+
+    @Inject
+    lateinit var viewModel: SafeSelectionViewModel
+
+    @Inject
+    lateinit var adapter: SafeSelectionAdapter
+
+    var navHandler: SafeOverviewNavigationHandler? = null
```

Comment:
Is this what you were referring as duplicate code? We could somehow refactor this into a helper class "ActiveSafeHelper" but, that's something we could do in a more tech debt oriented sprint.